runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyServletEngineAdapter.java

```diff
@@ -113,19 +118,54 @@ public void start(String serverInfo, ServletEngineAdapter.Config runtimeOptions)
       server.setHandler(appVersionHandlerMap);
     }
 
-    if (runtimeOptions.useJettyHttpProxy()) {
-      server.setAttribute(
-          "com.google.apphosting.runtime.jetty9.appYaml",
-          appYaml.orElseGet(() -> JettyServletEngineAdapter.getAppYaml(runtimeOptions)));
-      JettyHttpProxy.startServer(runtimeOptions);
-    }
-
     try {
-      server.start();
-    } catch (Exception ex) {
-      // TODO: Should we have a wrapper exception for this
-      // type of thing in ServletEngineAdapter?
-      throw new RuntimeException(ex);
+      boolean startJettyHttpProxy = false;
+      if (runtimeOptions.useJettyHttpProxy()) {
+        AppInfoFactory appInfoFactory;
+        AppVersionKey appVersionKey;
+        /* The init actions are not done in the constructor as they are not used when testing */
+        String appRoot = runtimeOptions.applicationRoot();
+        String appPath = runtimeOptions.fixedApplicationPath();
+        appInfoFactory = new AppInfoFactory(System.getenv());
+        AppinfoPb.AppInfo appinfo = appInfoFactory.getAppInfoFromFile(appRoot, appPath);
+        // TODO Should we also call ApplyCloneSettings()?
+        LocalRpcContext<EmptyMessage> context = new LocalRpcContext<>(EmptyMessage.class);
+        EvaluationRuntimeServerInterface evaluationRuntimeServerInterface =
+                Objects.requireNonNull(runtimeOptions.evaluationRuntimeServerInterface());
+        evaluationRuntimeServerInterface.addAppVersion(context, appinfo);
+        context.getResponse();
```

Comment:
need to add a unused variable there to avoid internal syle error

Reply:
done